pydmd/costs.py

```
@@ -348,14 +407,33 @@ def build_kern(window_length):
 
         :param window_length: Length of the data window in units of time
         :type window_length: int
+        :param relative_filter_length: A parameter governing how strongly
+        weighted the windowed construction is. Larger values mean more
+        strongly weighting the middle of the window.
+        :type relative_filter_length: float
+        :param direction: Specify the special cases for reconstructing
+        windows at the beginning and end of the time domain.
+        :type direction: string
         :return: Gaussian filter of length `window_length`
         :rtype: np.ndarray
+
         """
-        recon_filter_sd = window_length / 8
+        recon_filter_sd = window_length / relative_filter_length
         recon_filter = np.exp(
             -((np.arange(window_length) - (window_length - 1) / 2) ** 2)
             / recon_filter_sd**2
         )
+        # Do not apply the kerning at the end of the time domain. This
+        # assists in fitting the last window of the decomposition and stops
+        # the propagation of errors at the edge of the time domain.
+        if direction == "forward":
+            recon_filter[(window_length // 2) :] = 1
+        # Do not apply the kerning at the beginning of the time domain. This
+        # assists in fitting the last window of the decomposition and stops
+        # the propagation of errors at the edge of the time domain.
+        elif direction == "backward":
```

Should we throw an exception if `direction` is anything else than those two possibilities?

---

I'm happy to include this, but I was not intending for this keyword to be exposed to the user. Would that intent modify how we approach this?